src/runtime/runtime.cs

```diff
@@ -371,6 +379,43 @@ private static void InitializePlatformData()
             Machine = MType;
         }
 
+        /// <summary>
+        /// Initializes the architecture used within the python interpreter
+        ///
+        /// For various reasons the python interpreter often has a different
+        /// architecture to that of the machine. For example on a 64 bit Windows
+        /// platform the CPython interpreter is compiled with 32 bit longs.
+        /// This method will allow pythonnet to determine at runtime how big
+        /// python's longs are.
+        /// </summary>
+        private static void InitializePythonArch()
+        {
+            IntPtr structModule = PyImport_ImportModule("struct");
+            IntPtr calcsizeMethod = PyObject_GetAttrString(structModule, "calcsize");
+            IntPtr methodArgs = PyTuple_New(1);
+            IntPtr pString = PyString_FromString("P");
+
+            if(PyTuple_SetItem(methodArgs, 0, pString) != 0) 
+            {
+                PythonArchitecture = MachineType.Other;
+                return;
+            }
+
+            var result = PyLong_AsLong(PyObject_Call(calcsizeMethod, methodArgs, IntPtr.Zero));
+
+            switch(result){
+                case 4:
+                    PythonArchitecture = MachineType.i386;
+                    break;
+                case 8:
+                    PythonArchitecture = MachineType.x86_64;
+                    break;
+                default:
+                    PythonArchitecture = MachineType.Other;
+                    break;
+            }
```

Comment:
This is just not true.

1. On Windows (and possibly other systems) you will get 4 even when running in 64bit mode, that is what makes things complicated
2. ARMs exist

If we could always rely on `sizeof(void*) == sizeof(long)`, we could easily use `IntPtr` in places where `long` is used, but it's not that simple.

Also, I don't see why we would need to draw this information from Python, Python's architecture must always be the same as the arch .NET is running on.

We also have the information we need already: https://github.com/pythonnet/pythonnet/pull/497/files#diff-7e0cf65d55c3fae27decb3c07faf3e29

Reply:
Yes you are correct. Should be calcsize("L") ... but if you're happy that checking Is32Bit and IsWindows is sufficent then will rework the other methods and remove this.